### PR TITLE
Refactoring done and reduced some functions complexity and work-time

### PR DESCRIPTION
**Pull request by thanseefpp**
## Describe changes
I implemented/fixed _ to achieve _.

## Pre-requisites
Please ensure you have done the following:
- [done ] I have read the **CONTRIBUTING.md** document.
- [ not done] If my change requires a change to docs, I have updated the documentation accordingly.
- [ not done any integrations] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ tested ] I have added tests to cover my changes.

## Types of changes

- [ done ] Bug fix (non-breaking change which fixes an issue)
- [ functional changes ] New feature (non-breaking change which adds functionality)
- [ exceptional changes ] Breaking change (fix or feature that would cause existing functionality to change)
- [ done ] Other (add details above)

### CODE REVIEW

File: src/zenml/client.py [hunk 1]
```diff
@@ -556,11 +556,10 @@ def is_inside_repository(file_path: str) -> bool:
...
         """
-        repo_path = Client.find_repository()
-        if not repo_path:
-            return False
-
-        return repo_path in Path(file_path).resolve().parents
+        if repo_path := Client.find_repository():
+            return repo_path in Path(file_path).resolve().parents
+        else:
+             return False
```
Comment (fa9r) [inline src/zenml/client.py:562]: No need for else after return :) 
```suggestion
        return False
```

**Review by fa9r — APPROVED** [review]
Nice, looks good to me!

**Comment by fa9r** [comment]
I ran formatting / linting script and pushed the respective fixes on your develop branch, not sure why they don't show up in this PR though

File: src/zenml/client.py (hunk 1 above)
Comment (thanseefpp) [reply src/zenml/client.py]: okay

**Comment by thanseefpp** [comment]
Screenshot 2023-07-29 at 1 30 10 PM

It shows there is an issue to merge with the b5f37b7 branch which is not changed by myself.

**Comment by fa9r** [comment]
> Screenshot 2023-07-29 at 1 30 10 PM
> It shows there is an issue to merge with the [b5f37b7](https://github.com/zenml-io/zenml/commit/b5f37b765effcdb6746d82d060eb7cbac77ea23e) branch which is not changed by myself.

Yeah you can ignore that, no worries

**Comment by fa9r** [comment]
CI already passed on your last commit so I think this should be ready to merge 🙌 

@thanseefpp again big thanks for the PR, really appreciated 🙏

**Pull request merged**